### DISCUSSION TITLE
Getting Ready for 4.4 and the Godot Asset Store!

### DISCUSSION THREAD

**Discussion by GuilhermeGSousa** [GuilhermeGSousa/godot-motion-matching#80] [Announcements]
Hi everyone!

There's been a lot of exciting stuff happen since last release. The focus of this one has been to more closely integrate motion matching into the existing animation systems in Godot, namely the `AnimationTree`. Do do this, we made a "Motion Matching Animation Node", which should give users more flexibility to use motion matching in tandem with other animation techniques (state machines, blend trees and IK especially).
It required some engine level changes to do this, which is why this release will only work on [Godot 4.4 dev 7](https://godotengine.org/article/dev-snapshot-godot-4-4-dev-7/) and above.

Some other noteworthy changes include:
- The Bone Data feature has fixed and is now usable, with better data visualization on the Motion Matching Editor to go with it.
- Various improvements were made to the [motion matching demo](https://github.com/GuilhermeGSousa/godot-motion-matching-demo) thanks to @fire.
- Motion Matching now uses KD Trees to run queries, which results in a 20x-30x performance improvement when compared to the previous naive implementation.
- Various improvement to the Motion Matching Editor
- Bugfixes galore

## What's Changed
* Fix macos build by @rkalnins in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/7
* Setup ci by @GuilhermeGSousa in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/8
* Add artefact upload by @GuilhermeGSousa in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/19
* Rewrite CI scripts by @GuilhermeGSousa in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/22
* Add missing DebugDraw3D binaries by @GuilhermeGSousa in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/27
* Support motion matching for characters using `NavigationAgent3D` by @GuilhermeGSousa in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/25
* Fixes to trajectories from testing on UE motion matching data by @GuilhermeGSousa in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/36
* Goodbye MMAnimationPlayer! by @GuilhermeGSousa in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/37
* Fix trajectory generation to support jumping by @GuilhermeGSousa in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/38
* Update README.md by @fire in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/64
* Implement Motion Matching Animation Node by @GuilhermeGSousa in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/69
* Update bindings to lastest AnimationNodeExtension by @GuilhermeGSousa in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/77

## New Contributors
* @rkalnins made their first contribution in https://github.com/GuilhermeGSousa/godot-motion-matching/pull/7

**Full Changelog**: https://github.com/GuilhermeGSousa/godot-motion-matching/compare/0.1...0.2

<hr /><em>This discussion was created from the release <a href='https://github.com/GuilhermeGSousa/godot-motion-matching/releases/tag/0.2'>Getting Ready for 4.4 and the Godot Asset Store!</a>.</em>

**Comment by GeorgeS2019** [comment]
@GuilhermeGSousa

Which Godot v4.4-dev8? win64 to use to check the latest demo

I use Godot_v4.4-dev7_win64.exe, it load the project but fail to run

**Reply by GuilhermeGSousa** [reply]
@GeorgeS2019 I'd ask you to open an issue if you're running into issues using the plugin. You should be able to use it on Godot 4.4-dev 7, there was an issue about this just prior to this release: https://github.com/GuilhermeGSousa/godot-motion-matching/issues/79

**Reply by GeorgeS2019** [reply]
@GuilhermeGSousa 
Why the Godot Editor exe created by Github action is the not latest (your working version of ) Godot Editor.exe that could run the demo?
https://github.com/GuilhermeGSousa/godot

**Reply by GuilhermeGSousa** [reply]
I don't know what issue you're running into, but you should probably be using this tagged version of Godot when testing this out. Its very possible that that fork isn't up to date and is missing the required changes.

**Comment by GeorgeS2019** [comment]
Coordinating with other Godot4 animation libraries
https://github.com/NeonfireStudio/godot-advanced-third-person-controller
https://github.com/Kaiden-z/Godot-Procedural-Animation-Tool

**Comment by GeorgeS2019** [comment]
@GuilhermeGSousa 
is the latest 4.4 beta 1 compatible with the latest version of this demo?
